basic.concepts.md

```diff
@@ -47,7 +47,7 @@ There are three basic design goals that we should strive for in a program:
 
 1. Try to save time (Time complexity).
 2. Try to save space (Space complexity).
-3. Try to have face.
+3. Try to save face.
```

Comment:
Is this meant to be have or save?

Reply:
It should be "save face" (preventing the program from locking up or generating reams of garbled data)